vue-app/src/components/AddToCartButton.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+  <div>
+    <form action="#" v-if="!inCart">
+      <div class="input-button">
+        <img style="margin-left: 0.5rem;" height="24px" src="@/assets/dai.svg">
+        <input
+          class="input"
+          name="amount"
+          autocomplete="on"
+          onfocus="this.value=''"
+          v-bind="$attrs"
+          :placeholder="defaultAmount"
+          v-model="amount"
+        >
+        <input
+          type="submit"
+          class="donate-btn"
+          value="Add to cart"
+          :disabled="!canContribute()"
+          @click.prevent="handleSubmit"
+        >
+      </div>
+    </form>
+    <div class="input-button" v-if="inCart">
+      <button class="donate-btn-full">
+        <span>In cart 🎉</span>
+      </button>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Prop } from 'vue-property-decorator'
+import { DateTime } from 'luxon'
+
+import { SAVE_CART } from '@/store/action-types'
+import { ADD_CART_ITEM } from '@/store/mutation-types'
+import { DEFAULT_CONTRIBUTION_AMOUNT } from '@/api/contributions'
+import { User } from '@/api/user'
+import { Project } from '@/api/projects'
+import { RoundStatus } from '@/api/round'
+import { CartItem } from '@/api/contributions'
+import ConnectionModal from '@/components/ConnectionModal.vue'
+
+@Component({
+  components: {
+    ConnectionModal,
+  },
+  inheritAttrs: false,
```

Comment:
Read this but not sure I understand 😄  https://vuejs.org/v2/api/#inheritAttrs

Mind explaining the purpose of this? I see you're passing `$attrs` to the input element.

Reply:
By default, Vue is going to spread all the attributes you pass to the component, to its root element (the one you have defined in your template).

`inheritAttrs: false` is going to tell Vue to stop doing that. Setting that to `false` you will be able to control exactly where you want to put the attributes in your template. You can do that, using the `$attrs` variable.

In this case, all the attributes that consumers want to give the consumer are going to be passed to the input, not the root div element.

```
// In this case, you will be able to do something like this

<add-to-cart-button autocomplete="off" disabled="true" ... />

// Those attrs are going to be spread in the inner input element, not in the root element
```

***

Now, talking about my implementation. Thanks for bringing this up! because I think we should remove this and leave the default behavior.

Why? because this was something I was working on before the current solution and I forgot to clean up honestly. The previous approach I was taking was a little bit more abstract and I was thinking of passing different attrs to the inner input. But in this final implementation, we not using it (at least at the moment), so I want to keep it simple and avoid adding extra noise.

Reply:
Cool! Thanks for the explanation. Good feature to know about 👍